src/Engine/Audio/SteamAudioInterface.cpp

```diff
@@ -1,27 +1,41 @@
 #include "SteamAudioInterface.hpp"
 #include <assert.h>
 
+using namespace Audio;
+
 SteamAudioInterface::SteamAudioInterface() {
     simSettings.sceneType = IPL_SCENETYPE_PHONON;
     simSettings.numRays = 12400;
-    simSettings.numDiffuseSamples = 32;
-    simSettings.numBounces = 1;
+    simSettings.numDiffuseSamples = 1024;
+    simSettings.numBounces = 3;
     simSettings.irDuration = 1.0;
-    simSettings.ambisonicsOrder = 0;
-    simSettings.maxConvolutionSources = 5;
+    simSettings.ambisonicsOrder = 1;
+    simSettings.maxConvolutionSources = 10;
     context = IPLContext{ nullptr, nullptr, nullptr };
 }
 
 SteamAudioInterface::~SteamAudioInterface() {
-
+    if (environment)
+        iplDestroyEnvironment(&environment);
+    if (scene)
+        iplDestroyScene(&scene);
 }
 
 void SteamAudioInterface::CreateScene(uint32_t numMaterials) {
+    if (scene) {
+        iplDestroyScene(&scene);
+        scene = NULL;
+    }
     IPLerror err = iplCreateScene(context, NULL, simSettings, numMaterials, &scene);
 }
 
 void SteamAudioInterface::FinalizeScene(IPLFinalizeSceneProgressCallback progressCallback) {
-    iplFinalizeScene(scene, NULL);
+    if (scene)
+        iplFinalizeScene(scene, NULL);
+}
+
+void SteamAudioInterface::CreateRenderers(SteamAudioRenderers*& renderers) {
+    renderers = new SteamAudioRenderers(environment, sAudio.envRenderer, sAudio.binauralRenderer);
 }
 
 SteamAudioInterface::SaveData SteamAudioInterface::SaveFinalizedScene() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'SaveFinalizedScene' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)